src/internal_modules/roc_core/list.h

```diff
@@ -158,7 +125,9 @@ class List : public NonCopyable<> {
     //! @pre
     //!  @p element should not be member of any list.
     void push_back(T& element) {
-        insert_(element, &head_);
+        impl_.push_back(element.list_node_data());
+
+        OwnershipPolicy<T>::acquire(element);
```

Comment:
```
impl_.insert(element.list_node_data(), &impl_.head());
```

this should work (if you'll return a reference from head()) - and impl.push_back won't be needed.